src/main/java/org/jnode/fs/xfs/directory/LeafHeader.java

```diff
@@ -1,16 +1,63 @@
 package org.jnode.fs.xfs.directory;
 
+import lombok.Getter;
 import org.jnode.fs.xfs.XfsObject;
 
 import java.io.IOException;
 
 /**
  * A XFS leaf info.
+ * It is either
+ * <pre>
+ *     typedef struct xfs_dir2_leaf_hdr {
```

Comment:
What do you think of the way I implemented the different versions of `INode`? Do you think it's worth implementing similar structure here and for `BlockDirectoryEntry`? It think it would be cleaner instead of having version checks in the functions to determine what values and offsets to read from the buffer.

Reply:
Good idea, I'll do it. Thanks.